### ISSUE TITLE
Refactor types library

### ISSUE THREAD

**Issue by Soremwar** [denoland/deno#4712]
The following issue tracks the refactoring of the std/types library.

This is made with the purpose of making this library more suitable for use in production now that 1.0 is so close. It introduces version tracking individual library tests and JSDoc documentation to be used along with `deno doc`.

- [x] Refactor current types library (Group by lib name and version). #4713
- [ ] Document current types to make them usable through `deno doc`. (Waiting on #4711)
- [ ] Add tests for required libraries.
- [ ] Include guidelines for contributing to this library in the Deno manual.

**Comment by axetroy** [comment]
BTW. `react-dom/server.d.ts` still not exist.

And this is important for server-side rendering

**Comment by Soremwar** [comment]
@axetroy Follow #4722

**Comment by Soremwar** [comment]
@kitsonk Can you assign me this issue too please? :sweat_smile:

**Comment by Soremwar** [comment]
Closing in favor of https://github.com/Soremwar/deno_types/issues/1
